File: python/Consommateur.py
from threading import Thread
import time

class Consommateur(Thread) :

    def __init__(self, queue, temps, label, num_consommateur) :
        Thread.__init__(self)
        self.queue = queue
        self.temps = temps
        self.label = label
        self.num_consommateur = num_consommateur
        self.interrupted = False

    def setInterrupted(self,bool) :
        self.interrupted = bool

    def run(self) :
        while not self.interrupted :
            self.label.config(text="Consommateur"+str(self.num_consommateur)+" : Retrait d'un entier en cours...")
            n = self.queue.defiler()
            self.label.config(text="Consommateur"+str(self.num_consommateur)+" : retrait de l'entier "+str(n)+" effectue.")
            time.sleep(self.temps/1000)

File: java/Consommateur.java
import java.awt.*;
import java.awt.event.*;
import javax.swing.*;

public class Consommateur extends Thread{

  private Queue queue;
  private int temps, num_consommateur;
  private JLabel label;

  public Consommateur(Queue queue, int temps, JLabel label, int num_consommateur){
    this.queue = queue;
    this.temps = temps;
    this.label = label;
    this.num_consommateur = num_consommateur;
  }

  public void run(){
    try{
      while(!interrupted()){
        this.label.setText("Consommateur"+this.num_consommateur+" : Retrait d'un entier en cours...");
        int n = this.queue.defiler();
        this.label.setText("Consommateur"+this.num_consommateur+" : retrait de l'entier "+n+" effectue.");
        sleep(this.temps);
      }
    }catch(Exception e){}
  }

}
